packages/lodestar-utils/src/json.ts

```diff
@@ -11,9 +20,16 @@ export function toJson(arg: unknown): Json {
     case "object":
       if (arg === null) return "null";
       if (arg instanceof Uint8Array) return toHexString(arg);
-      if (arg instanceof LodestarError) return toJson(arg.toObject());
+      if (arg instanceof LodestarError) {
+        const copyArg = arg.toObject();
+        convertArrayBufferViews(copyArg);
+        return toJson(copyArg);
+      }
       if (arg instanceof Error) return toJson(errorToObject(arg));
-      return arg as Json;
+      if (arg instanceof Array) return arg as Json;
+      copyArg = Object.assign({}, arg);
```

Comment:
It's hard for me to follow what's going on here. Could you describe the purpose of these statements and the global untyped `copyArg` variable?

Reply:
I didn't want to have to modify the argument of the function, so I made a copy of it.

Reply:
Can you abstract this functionality in a way that you don't need to mutate arguments out of the scope of the if statement?

Reply:
Yes, my latest commit (0cc57106367b0a0158f4a944a25d195fce00c3d9) should address this and your other comments.  Let me know what you think.